src/VisualStudio/Core/Def/Implementation/ProjectSystem/RuleSets/VisualStudioRuleSetManager.RuleSetFile.cs

```diff
@@ -92,51 +104,76 @@ public ReportDiagnostic GetGeneralDiagnosticOption()
 
             private void EnsureSubscriptions()
             {
-                if (!_subscribed)
+                lock (_gate)
                 {
-                    foreach (var tracker in _trackers)
+                    if (!_subscribed)
                     {
-                        tracker.EnsureSubscription();
-                    }
+                        foreach (var tracker in _trackers)
+                        {
+                            tracker.EnsureSubscription();
```

Comment:
💭 This is another call inside a lock that makes me uncomfortable

Reply:
Thinking about it again, I think we're good. In some sense this problem already existed; it was being called in the context of the lock known as the UI thread.